packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -2056,6 +2176,148 @@ class _BackLabel extends StatelessWidget {
   }
 }
 
+/// The 'Cancel' button next to the search field in a
+/// [CupertinoSliverNavigationBar.search].
+class _CancelButton extends StatelessWidget {
+  const _CancelButton({this.opacity = 1.0, required this.onPressed});
+
+  final void Function()? onPressed;
+  final double opacity;
+
+  @override
+  Widget build(BuildContext context) {
+    return Align(
+      alignment: Alignment.centerLeft,
+      child: Opacity(
+        opacity: opacity,
+        child: CupertinoButton(
+          padding: EdgeInsets.zero,
+          onPressed: onPressed,
+          // TODO(victorsanni): Localize this string.
```

Comment:
Hi, I created [PR #162581](https://github.com/flutter/flutter/pull/162581), which fixes this.